Nodejs/Product/TestAdapter/ProjectTestDiscoverer.cs

```diff
@@ -12,11 +12,12 @@
 
 namespace Microsoft.NodejsTools.TestAdapter
 {
-    // Keep in sync the method TypeScriptHelpers.IsSupportedTestProjectFile if there's a change on the supported projects.
-    [FileExtension(NodejsConstants.NodejsProjectExtension), FileExtension(NodejsConstants.CSharpProjectExtension), FileExtension(NodejsConstants.VisualBasicProjectExtension), FileExtension(NodejsConstants.JavaScriptProjectExtension)]
-    [DefaultExecutorUri(NodejsConstants.ExecutorUriString)]
-    public partial class ProjectTestDiscoverer : ITestDiscoverer
+    // We require to put a non-existent file extension to avoid duplicate discovery executions.
+    [FileExtension("NTVS_NonExistentFileExtension")]
+    public class ProjectTestDiscoverer : ITestDiscoverer
     {
+        public virtual string TestDiscovererName => nameof(ProjectTestDiscoverer);
```

Comment:
`abstract`?

Reply:
Same as above, the class is instantiated so we declare the name for it. Inheritors replace it value, hence the virtual.

Reply:
My personal preference would be to make this type abstract and make a new concrete type that extends it.  More generally, it's easier to reason about type hierarchies if everything is either abstract or sealed.

Reply:
I don't mind one or another, so I change das suggested. I considered it before but decided to have the lease amount of classes was better for maintainability, but I do agree that having explicit abstract and sealed classes.

The new class name is `DefaultTestDiscoverer` because I'm not 100% sure when it is called. I have only seen it when using the vstest cli, but the documentation doesn't state exactly in what conditions it's called: "Runs 'all' the tests present in the specified 'sources'.". Let me know if you have a suggestion on the name.

Reply:
"Default" seems good.  Thanks!